File: frontend/src/app/models/account.ts
export class Account {
  public public_id?: string;
  public first_name?: string;
  public last_name?: string;
  public email?: string;
  public role?: string;
  public jwtToken?: string;
  public refreshToken?: string;
  /**
   *
   */
  constructor (id?: string, firstName?: string, lastName?: string, email?: string, role?: string, jwtToken?: string) {
    this.public_id = id;
    this.first_name = firstName;
    this.last_name = lastName;
    this.email = email;
    this.role = role;
    this.jwtToken = jwtToken;
  }

  static Create() {
    return this;
  }
}

File: dotnet_core/lan-management-api/Entities/Account.cs
namespace lan_management_api.Entities;

public class Account
{
    [Key]
    public int Id { get; set; }
    [MaxLength(50)]
    public string Title { get; set; }
    [MaxLength(50)]
    public string FirstName { get; set; }
    [MaxLength(50)]
    public string LastName { get; set; }
    [MaxLength(50)]
    public string Email { get; set; }
    [MaxLength(64)]
    public string PasswordHash { get; set; }
    public bool AcceptTerms { get; set; }
    public Role Role { get; set; }
    [MaxLength(128)]
    public string VerificationToken { get; set; }
    public DateTime? Verified { get; set; }
    public bool IsVerified => Verified.HasValue || PasswordReset.HasValue;
    [MaxLength(64)]
    public string ResetToken { get; set; }
    public DateTime? ResetTokenExpires { get; set; }
    public DateTime? Created { get; set; }
    public DateTime? Updated { get; set; }
    public List<RefreshToken> RefreshTokens { get; set; }
    public DateTime? PasswordReset { get; set; }
    public bool OwnsToken(string token)
    {
        return RefreshTokens?.Find(x => x.Token == token) != null;
    }
}